File: YeProfspilka.Admin/src/app/models/DiscountType.ts
import { Option } from './ui-models/Option';

export enum DiscountType {
	AvailableForAll = 0,
	AvailableForMemberOfProf = 1,
	OneTimeForAll = 2,
	OneTimeForMemberOfProf = 3,
	OneTimeForFirstCourse = 4,
}

export const DiscountTypeOptions: Option[] = [
	{ label: 'Доступна для всіх', value: DiscountType.AvailableForAll },
	{ label: 'Доступна для членів профспілки', value: DiscountType.AvailableForMemberOfProf },
	// { label: 'Одноразова для всіх', value: DiscountType.OneTimeForAll },
	// { label: 'Одноразова для членів профспілки', value: DiscountType.OneTimeForMemberOfProf },
	// { label: 'Доступна для першого курсу', value: DiscountType.OneTimeForFirstCourse },
];

File: src/api/EProfspilka.Core/Enumerations/DiscountType.cs
namespace EProfspilka.Core.Enumerations;

[Flags]
public enum DiscountType
{
    None = 0,                          // 0
    AvailableForAll = 1 << 0,          // 1
    AvailableForMembers = 1 << 1,      // 2
    OneTimeForAll = 1 << 2,            // 4
    OneTimeForMembers = 1 << 3,        // 8
    OneTimeForFirstCourse = 1 << 4,    // 16
}